.github/workflows/sql-test-and-build-workflow.yml

```
@@ -19,26 +19,35 @@ on:
 
 jobs:
   build:
+    env:
+      BUILD_ARGS: ${{ matrix.entry.os_build_args }}
     strategy:
+      # Run all jobs
+      fail-fast: false
       matrix:
-        java:
-          - 11
-          - 17
-    runs-on: ubuntu-latest
+        entry:
+          - { os: ubuntu-latest, java: 11 }
+          - { os: windows-latest, java: 11, os_build_args: -x doctest -x integTest -x jacocoTestReport -x compileJdbc}
+          - { os: macos-latest, java: 11, os_build_args: -x doctest -x integTest -x jacocoTestReport -x compileJdbc }
+          - { os: ubuntu-latest, java: 17 }
+          - { os: windows-latest, java: 17, os_build_args: -x doctest -x integTest -x jacocoTestReport -x compileJdbc }
+          - { os: macos-latest, java: 17, os_build_args: -x doctest -x integTest -x jacocoTestReport -x compileJdbc }
+    runs-on: ${{ matrix.entry.os }}
 
     steps:
     - uses: actions/checkout@v3
-    
+
     - name: Set up JDK ${{ matrix.java }}
       uses: actions/setup-java@v3
       with:
         distribution: 'temurin'
-        java-version: ${{ matrix.java }}
-    
+        java-version: ${{ matrix.entry.java }}
+
     - name: Build with Gradle
-      run: ./gradlew --continue build assemble
+      run: ./gradlew --continue build ${{ env.BUILD_ARGS }}
```

I think you can use there
```suggestion
      run: ./gradlew --continue build ${{ matrix.entry.os_build_args }}
```

---

You're right this is cleaner let me do this